spec/lib/vmdb/settings_spec.rb

```diff
@@ -293,54 +293,32 @@
-          expect(Vmdb::Settings.for_resource(miq_server).api.new_key).to eq "new value"
-          expect(Vmdb::Settings.for_resource(second_server).api.token_ttl).to eq server_value
+          expect(described_class.for_resource(miq_server).api.new_key).to eq "new value"
+          expect(described_class.for_resource(second_server).api.token_ttl).to eq server_value
         end
 
         it "deletes current node and all sub-nodes for the resource if specified on node level" do
           described_class.save!(miq_server, :api => reset)
 
           expect(SettingsChange.count).to eq 6
           expect(miq_server.settings_changes.where("key LIKE ?", "/api%").count).to eq 0
-          expect(Vmdb::Settings.for_resource(miq_server).api.new_key).to eq nil
-          expect(Vmdb::Settings.for_resource(second_server).api.token_ttl).to eq server_value
+          expect(described_class.for_resource(miq_server).api.new_key).to eq nil
+          expect(described_class.for_resource(second_server).api.token_ttl).to eq server_value
         end
 
         it "deletes new key-value settings not present in defaul yaml" do
           described_class.save!(miq_server, :api => {:new_key => reset})
-          expect(Vmdb::Settings.for_resource(miq_server).api.new_key).to eq nil
+          expect(described_class.for_resource(miq_server).api.new_key).to eq nil
         end
 
         it "deletes array" do
           described_class.save!(miq_server, :array => reset)
-          expect(Vmdb::Settings.for_resource(miq_server).array).to be nil
+          expect(described_class.for_resource(miq_server).array).to be nil
         end
 
         it "passes validation" do
           described_class.save!(miq_server, :session => {:timeout => reset})
-          expect(Vmdb::Settings.for_resource(miq_server).session.timeout).to eq zone_value
-        end
-      end
-
-      context "magic value <<reset_all>>" do
-        it "deletes all key-value for specific key for all resource if specified on leaf levelher" do
-          described_class.save!(miq_server, :api => {:token_ttl => reset_all})
-
-          expect(SettingsChange.where("key LIKE ?", "/api/token_ttl").count).to eq 0
-          expect(SettingsChange.where("key LIKE ?", "/api%").count).to eq 1
-          expect(SettingsChange.where("key LIKE ?", "/session%").count).to eq 3
-        end
-
-        it "deletes specific node and all sub-nodes for all resources if specified on node level" do
-          described_class.save!(miq_server, :api => reset_all)
-
-          expect(SettingsChange.where("key LIKE ?", "/api%").count).to eq 0
-          expect(SettingsChange.where("key LIKE ?", "/session%").count).to eq 3
-        end
-
-        it "passes validation" do
-          described_class.save!(miq_server, :session => {:timeout => reset_all})
-          expect(SettingsChange.where("key LIKE ?", "/session%").count).to eq 0
+          expect(described_class.for_resource(miq_server).session.timeout).to eq zone_value
```

Comment:
This spec actually looks exactly like what I would expect `"deletes key-value for specific key for the resource if specified on leaf level"` to look like.